lib/Drupal/service_container/Tests/LegacyMessengerTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\service_container\Tests\LegacyMessengerTest.
+ */
+
+namespace Drupal\service_container\Tests;
+
+use Drupal\service_container\Messenger\LegacyMessenger;
+
+class LegacyMessengerTest extends ServiceContainerIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getInfo() {
+    return array(
+      'name' => 'LegacyMessenger',
+      'description' => 'Tests the messenger integration',
+      'group' => 'service_container',
+    );
+  }
+
+  public function testLegacyMessenger() {
+    /** @var \Drupal\service_container\Messenger\LegacyMessenger $messenger_service */
+
+    $messenger_service = $this->container->get('messenger');
+    $this->assertTrue($messenger_service instanceof LegacyMessenger);
```

Comment:
Would love some helpful explanations for the assertions, also there is assertInstanceOf().

e.g.

```php
$this->assertInstanceOf('Drupal\service_container\Messenger\LegacyMessenger', 'Messenger is instanceof LegacyMessenger');
```